--- a/net/minecraft/client/renderer/tileentity/TileEntityRenderer.java
+++ b/net/minecraft/client/renderer/tileentity/TileEntityRenderer.java
@@ -17,7 +17,7 @@
 import net.minecraftforge.api.distmarker.OnlyIn;
 
 @OnlyIn(Dist.CLIENT)
-public abstract class TileEntityRenderer<T extends TileEntity> {
+public abstract class TileEntityRenderer<T extends TileEntity> implements net.minecraftforge.client.model.dynamic.adapters.ITileEntityRenderable<T> {
    public static final ResourceLocation[] field_178460_a = new ResourceLocation[]{new ResourceLocation("textures/" + ModelBakery.field_207770_h.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207771_i.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207772_j.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207773_k.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207774_l.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207775_m.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207776_n.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207777_o.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207778_p.func_110623_a() + ".png"), new ResourceLocation("textures/" + ModelBakery.field_207779_q.func_110623_a() + ".png")};
    protected TileEntityRendererDispatcher field_147501_a;
 
@@ -66,6 +66,8 @@
       return false;
    }
 
+   public void renderTileEntityFast(T te, double x, double y, double z, float partialTicks, int destroyStage, net.minecraft.client.renderer.BufferBuilder buffer) {}
+
    protected void func_190052_a(T p_190052_1_, String p_190052_2_, double p_190052_3_, double p_190052_5_, double p_190052_7_, int p_190052_9_) {
       ActiveRenderInfo activerenderinfo = this.field_147501_a.field_217666_g;
       double d0 = p_190052_1_.func_145835_a(activerenderinfo.func_216785_c().field_72450_a, activerenderinfo.func_216785_c().field_72448_b, activerenderinfo.func_216785_c().field_72449_c);
