--- a/net/minecraft/network/play/server/SEntityMetadataPacket.java
+++ b/net/minecraft/network/play/server/SEntityMetadataPacket.java
@@ -12,29 +12,42 @@
 public class SEntityMetadataPacket implements IPacket<IClientPlayNetHandler> {
    private int field_149379_a;
    private List<EntityDataManager.DataEntry<?>> field_149378_b;
+   private @javax.annotation.Nullable PacketBuffer capabilityData;
+   public PacketBuffer getCapabilityData() { return this.capabilityData; }
 
    public SEntityMetadataPacket() {
    }
 
-   public SEntityMetadataPacket(int p_i46917_1_, EntityDataManager p_i46917_2_, boolean p_i46917_3_) {
-      this.field_149379_a = p_i46917_1_;
-      if (p_i46917_3_) {
-         this.field_149378_b = p_i46917_2_.func_187231_c();
-         p_i46917_2_.func_187230_e();
+   public SEntityMetadataPacket(int entityIdIn, EntityDataManager dataManagerIn, boolean sendAll, PacketBuffer capabilityData) {
+      this.field_149379_a = entityIdIn;
+      if (sendAll) {
+         this.field_149378_b = dataManagerIn.func_187231_c();
+         dataManagerIn.func_187230_e();
       } else {
-         this.field_149378_b = p_i46917_2_.func_187221_b();
+         this.field_149378_b = dataManagerIn.func_187221_b();
       }
-
+      this.capabilityData = capabilityData;
    }
+   // Used for vanilla packet filter
+   public SEntityMetadataPacket(int entityIdIn, List<EntityDataManager.DataEntry<?>> dataManagerEntries) {
+       this.field_149379_a = entityIdIn;
+       this.field_149378_b = dataManagerEntries;
+       this.capabilityData = null;
+   }
 
    public void func_148837_a(PacketBuffer p_148837_1_) throws IOException {
       this.field_149379_a = p_148837_1_.func_150792_a();
       this.field_149378_b = EntityDataManager.func_187215_b(p_148837_1_);
+      this.capabilityData = new PacketBuffer(p_148837_1_.readBytes(p_148837_1_.func_150792_a()));
    }
 
    public void func_148840_b(PacketBuffer p_148840_1_) throws IOException {
       p_148840_1_.func_150787_b(this.field_149379_a);
       EntityDataManager.func_187229_a(this.field_149378_b, p_148840_1_);
+      if (this.capabilityData != null) { // Only null for vanilla clients!
+         p_148840_1_.func_150787_b(this.capabilityData.readableBytes());
+         if (this.capabilityData.isReadable()) p_148840_1_.writeBytes(this.capabilityData);
+      }
    }
 
    public void func_148833_a(IClientPlayNetHandler p_148833_1_) {
